File: JavaScript_Basic_04_2023_SolutionsOfProblems/04.Ex_PB-JS-For-Loop/P06_Oscars.js
function oscars(input) {
    let nameOfArtist = input[0];
    let points = Number(input[1]);
    let numberOfJury = Number(input[2]);

    for (let i = 0; i < numberOfJury; i++) {
        points += input[i * 2 + 3].length * input[i * 2 + 4] / 2;
        if (points > 1250.5) {
            console.log(`Congratulations, ${nameOfArtist} got a nominee for leading role with ${points.toFixed(1)}!`);
            break;
        }
    }
    if (!(points > 1250.5)) {
        console.log(`Sorry, ${nameOfArtist} you need ${(1250.5 - points).toFixed(1)} more!`);
    }

}
oscars(["Zahari Baharov",
    "205",
    "4",
    "Johnny Depp",
    "45",
    "Will Smith",
    "29",
    "Jet Lee",
    "10",
    "Matthew Mcconaughey",
    "39"])

File: Java_Basic_Jan_2023_SolutionsOfProblems/P04Ex_FOR_Loop/P06_Oscars.java
package P04Ex_FOR_Loop;

import java.util.Scanner;

public class P06_Oscars {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String nameArtist = scanner.nextLine();
        String nameJury = "";
        double pointsJury = 0;
        double points = Double.parseDouble(scanner.nextLine());
        int numberJury = Integer.parseInt(scanner.nextLine());
        for (; numberJury > 0 && points <= 1250.5; numberJury--) {
            nameJury = scanner.nextLine();
            pointsJury = Double.parseDouble(scanner.nextLine());
            points += nameJury.length() * pointsJury / 2;
        }
        if (points > 1250.5) {
            System.out.printf("Congratulations, %s got a nominee for leading role with %.1f!"
                    , nameArtist, points);
        } else {
            System.out.printf("Sorry, %s you need %.1f more!", nameArtist, 1250.5 - points);
        }
    }
}
